src/electron/index.cjs

```diff
@@ -1,37 +1,67 @@
-const { app, BrowserWindow, ipcMain } = require("electron");
+const { app, BrowserWindow, ipcMain, Notification } = require("electron");
 const path = require("path");
-const { saveUserData, readUserData } = require("./file-io.cjs");
+const {
+  saveUserData,
+  readUserData,
+  saveTerminalOutput,
+} = require("./file-io.cjs");
+let child_process = require("child_process");
 
 function createWindow() {
   const win = new BrowserWindow({
     width: 1050,
     height: 605,
     webPreferences: {
-      preload: path.join(__dirname, 'preload.js')
-    }
+      preload: path.join(__dirname, "preload.js"),
+    },
   });
 
   if (process.env.NODE_ENV !== "development") {
     // Load production build
     win.loadFile(`${__dirname}/../svelte/dist/index.html`);
 
     //hide menu in production
-    win.setMenu(null)
+    win.setMenu(null);
   } else {
     // Load vite dev server page
     win.loadURL("http://localhost:5173/");
   }
 }
 
-ipcMain.on('saveMacros', (event, macros) => {
+ipcMain.on("saveMacros", (event, macros) => {
   saveUserData(macros);
 });
+ipcMain.on("execute-command", (event, command) => {
+  // '\n' has to be filtered out because when using '^' on multiple lines, the process is in an infinite loop
```

Comment:
"the process is in an infinite loop" Ist das denn korrekt? Endlosschleife heißt ja eigentlich, dass der ganze RAM recht schnell weg sein müsste, oder?